WalletWasabi.Backend/Controllers/WabiSabiController.cs

```diff
@@ -28,31 +31,27 @@ public Task<RoundState[]> GetStatusAsync(CancellationToken cancellationToken)
 		}
 
 		[HttpPost("connection-confirmation")]
-		[Idempotent]
-		public Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(ConnectionConfirmationRequest request, CancellationToken cancellableToken)
+		public Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(ConnectionConfirmationRequest request, CancellationToken cancellationToken)
 		{
-			return Arena.ConfirmConnectionAsync(request, cancellableToken);
+			return IdempotencyRequestCache.GetCachedResponseAsync(request, action: (request, token) => Arena.ConfirmConnectionAsync(request, token), cancellationToken);
```

Comment:
is there a technical reason i'm missing that this is done manually (as opposed to modifying the `Idempotent` attr to use the new behavior) or is this a stylistic choice?

Reply:
So I did it this way because I got the impression from https://github.com/zkSNACKs/WalletWasabi/issues/6257#issue-982152124 that it's what we want. The second reason is basically that I can test the code much more easily when it is not the C# attribute - this seems important because it's actually easy with different approaches to get to state "it seems the idempotent implementation works fine ... but sometimes it doesn't". 

We can still have the original implementation though. 

The new implementation seems to offer better paramerization options in the future, if we want something like that. I mean it's easier to do something more complex with the new impl because it's just standard C# code. Idempotent C# attribute allows that possibly too but once we get to the intersection of technologies (ASP.NET & our code), it seems it's much harder to reason about it because we don't have expertise in ASP.NET really and that leads us to [experiments like this](https://github.com/zkSNACKs/WalletWasabi/pull/6615#issuecomment-958117223) imho.

Reply:
:+1:

Personally I prefer the more explicit approach, because it's not like these are super complicated methods full of boilerplate, i like syntactic sugar like attrs when the indirection and requirement to be familiar with more mechanisms/concepts/abstractions makes a big difference to my cognitive load when reading. Since these are just wrappers around `Arena` I find the explicit form to be more readable even though it's more verbose